subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -468,10 +468,10 @@ u8_t ll_adv_enable(u8_t enable)
 
 		/* AdvA, fill here at enable */
 		if (h->adv_addr) {
-			/* TODO: Privacy */
 			u8_t *tx_addr = ll_addr_get(pdu_adv->tx_addr, NULL);
 
-			if (!mem_nz(tx_addr, BDADDR_SIZE)) {
+			/* TODO: Privacy check */
+			if (pdu_adv->tx_addr && !mem_nz(tx_addr, BDADDR_SIZE)) {
```

Comment:
This is in the advertising extensions block, is this what you want?

Reply:
This is reverting back to what was before, not to allow zeroes as random address, but allow if public.